### ISSUE TITLE
RecordBatch doesn't support indexing by int

### ISSUE THREAD

**Issue by tekumara** [zen-xu/pyarrow-stubs#40]
When using an integer item key on RecordBatch, eg: `batch[8]`: 

```
No overloads for "__getitem__" match the provided argumentsPylance[reportCallIssue](https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportCallIssue)
lib.pyi(1162, 9): Overload 2 is the closest match
Argument of type "Literal[8]" cannot be assigned to parameter "key" of type "_builtin_slice" in function "__getitem__"
  "Literal[8]" is incompatible with "slice"Pylance[reportArgumentType](https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportArgumentType)
 ```
  
 See [_Tabular.__get__item](https://github.com/apache/arrow/blob/e61c105c73dfabb51d5afc972ff21cc5326b3d93/python/pyarrow/table.pxi#L1634):
 
 > key : integer, str, or slice
  
 NB: RecordBatch inherits from _Tabular

**Comment by zen-xu** [comment]
Thank you very much. I am currently manually refactoring the entire project (instead of generating it with mypy). I will release it after some time.

**Comment by zen-xu** [comment]
resolved by #35
